### PR TITLE
#170932632 user advertiser update announcement endpoint ch3

### PR DESCRIPTION
**Pull request by the22mastermind**
#### What does this PR do?
Add advertiser update announcement functionality with database
#### How should this be manually tested?

- Clone this repo
- run `cd announceIT && npm run test`

#### Any background context you want to provide?
An advertiser will be able to update only the announcements he/she has created
#### What are the relevant pivotal tracker stories?
#170932632
#### Questions:

### CODE REVIEW

File: server/controllers/advertiserCh3.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+const createAnnouncement = async (req, res) => {
+  // Joi Validation
+  const { error } = validation.validateCreateAnnouncement(req.body);
+  if (error) {
+    return utils.returnError(res, codes.statusCodes.badRequest, error.details[0].message);
+  }
+  // Retrieve token info
+  const token = req.headers.authorization.split(' ')[1];
+  const decoded = jwt.verify(token, process.env.JWT_KEY);
+  req.userData = decoded;
```
Comment (alainburindi) [inline server/controllers/advertiserCh3.js]: this should be in a middleware

File: server/controllers/advertiserCh3.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+const updateAnnouncement = async (req, res) => {
+  // Joi Validation
+  const { error } = validation.validateUpdateAnnouncement(req.body);
+  if (error) {
+    return utils.returnError(res, codes.statusCodes.badRequest, error.details[0].message);
+  }
+  // Retrieve token info
+  const token = req.headers.authorization.split(' ')[1];
+  const decoded = jwt.verify(token, process.env.JWT_KEY);
+  req.userData = decoded;
```
Comment (alainburindi) [inline server/controllers/advertiserCh3.js]: consider the previous comment

File: server/utils/queries.js [hunk 3]
```diff
@@ -25,7 +25,60 @@ const signupUser = async (data) => {
...
+const saveAnnouncement = async (data) => {
+  const announcement = await pool.query('INSERT INTO announcements(title,text,start_date,end_date,status,owner,createdon) VALUES($1,$2,$3,$4,$5,$6,$7) RETURNING*',
+    [
+      data.title,
+      data.description,
+      data.startdate,
+      data.enddate,
+      data.status,
+      data.owner,
+      data.createdon,
```
Comment (alainburindi) [inline server/utils/queries.js:50]: you can just use the spreading operator

**Review by alainburindi — CHANGES_REQUESTED** [review]

File: server/utils/queries.js (hunk 3 above)
Comment (the22mastermind) [reply server/utils/queries.js:50]: It's throwing an error when I use spreading

**Pull request merged**